packages/init/main.tf

```diff
@@ -1,3 +1,66 @@
+
+# Enable Secrets Manager API
+resource "google_project_service" "secrets_manager_api" {
+  #project = var.gcp_project_id
+  service = "secretmanager.googleapis.com"
+
+  disable_on_destroy = false
+}
+
+# Enable Certificate Manager API
+resource "google_project_service" "certificate_manager_api" {
+  #project = var.gcp_project_id
+  service = "certificatemanager.googleapis.com"
+
+  disable_on_destroy = false
+}
+
+# Enable Compute Engine API
+resource "google_project_service" "compute_engine_api" {
+  #project = var.gcp_project_id
+  service = "compute.googleapis.com"
+
+  disable_on_destroy = false
+}
+
+# Enable Artifact Registry API
+resource "google_project_service" "artifact_registry_api" {
+  #project = var.gcp_project_id
+  service = "artifactregistry.googleapis.com"
+
+  disable_on_destroy = false
+}
+
+# Enable OS Config API
+resource "google_project_service" "os_config_api" {
+  #project = var.gcp_project_id
+  service = "osconfig.googleapis.com"
+
+  disable_on_destroy = false
+}
+
+# Enable Stackdriver Monitoring API
+resource "google_project_service" "monitoring_api" {
+  #project = var.gcp_project_id
+  service = "monitoring.googleapis.com"
+
+  disable_on_destroy = false
+}
+
+# Enable Stackdriver Logging API
+resource "google_project_service" "logging_api" {
+  #project = var.gcp_project_id
+  service = "logging.googleapis.com"
+
+  disable_on_destroy = false
+}
+
+resource "time_sleep" "secrets_api_wait_60_seconds" {
+  depends_on = [google_project_service.secrets_manager_api]
+
+  create_duration = "60s"
+}
+
```

Comment:
This is 80/20 improvement. Feel free to improve it by changing it to local exec, which would check if the api is already ready

Reply:
LGTM, I think it's useful to have and lean towards 80/20 as a rule. I reduced the time to 20s, When I ran through enabling the APIs they seemed to take about 10s at most, so we can split the difference.